### DISCUSSION TITLE
Is this library FIPS compliant?

### DISCUSSION THREAD

**Discussion by santhoshramaraju** [Yubico/libfido2#776] [Q&A] [answered]
Title says it all

**Comment by LDVG** [comment, marked as answer]
Hi,

libfido2 has not been FIPS certified.

**Reply by VedaviBalaji** [reply]
Can we detect if the key is FIPS and fail it accordingly? Basically any touch on my FIPS key `YubiKey FIPS (4.4.5) [OTP+FIDO]` is not detected.

**Reply by LDVG** [reply]
Hi,

What a relying party would typically do is to request direct attestation during the registration ceremony.  You would then use knowledge gained from the attestation object to verify the authenticator's make and model and, based on that, make a decision on whether to accept the credential or not.

I'll link some additional resources below:
- https://developers.yubico.com/WebAuthn/Concepts/Securing_WebAuthn_with_Attestation.html
- https://www.w3.org/TR/webauthn-3/#reg-ceremony-verify-attestation
